nussl/core/audio_signal.py

```diff
@@ -1644,6 +1649,311 @@ def embed_audio(self, ext='.mp3', display=True):
         from . import play_utils
         return play_utils.embed_audio(self, ext=ext, display=display)
 
+    ##################################
+    #          Effect Hooks          #
+    ##################################
+
+    def build_effects(self, reset=True, overwrite=False):
+        """
+        Builds all effects in the AudioSignal's effects chains using the audio data of the calling 
+        AudioSignal. If reset=True, the current effects chains will be deleted. If overwrite=True, 
+        the calling AudioSignal's audio_data will be overwritten and the stft_data will be deleted. 
+
+        FFmpeg effects will be applied AFTER SoX effects. The effects time_stretch and pitch_shift 
+        are SoX effects. All others are FFmpeg effects.
+
+        If no effects have been added to the AudioSignal's effects chains, then build_effects 
+        will throw a RuntimeError. 
+
+        This will add a tremolo effect followed by a high pass effect to the AudioSignal's effects chains:
+
+        >>> audio_signal.tremolo(5, .6).high_pass(12000)
+
+        You can apply then apply the effects onto the AudioSignal with build_effects
+
+        >>> audio_signal.build_effects(overwrite=True)
+
+        Args:
+            reset (bool): If True, clears out all effects in effect chains. Default=True
+            overwrite (bool): If True, overwrites existing audio_data in AudioSignal. Defult=False
+                Also clears out stft_data
+        Returns:
+            self or new_signal (AudioSignal): If overwrite=True, then returns initially AudioSignal 
+            with edited audio_data. Otherwise, returns a new AudioSignal new_signal.
+        """
+        if not self._ffmpeg_effects_chain and not self._sox_effects_chain:
+            raise RuntimeError("No effect hooks have been called on this AudioSignal")
```

Comment:
I don't think this should raise an error. If there are no effects, it should just `return self` without making a fuss.

Reply:
Removed Error